Base/QTCore/qSlicerCoreIOManager.cxx

```diff
@@ -880,34 +883,46 @@ bool qSlicerCoreIOManager::exportNodes(
           if (userMessages)
             {
             userMessages->AddMessage(vtkCommand::ErrorEvent,
-              (tr("Error encountered while exporting %1").arg(storableNode->GetName())).toStdString()
+              (tr("Error encountered while exporting %1.").arg(storableNode->GetName())).toStdString()
             );
             }
           success = false;
           continue;
           }
         compositeTransformNode->SetAndObserveTransformFromParent(generalTransform);
         temporaryNodeAsTransformable->SetAndObserveTransformNodeID(compositeTransformNode->GetID());
+        temporaryNodeAsTransformable->HardenTransform();
         }
-
-      temporaryNodeAsTransformable->HardenTransform();
       }
 
-
-    // Prepare temporary scene node ID and prepare filename for saving
+    // Copy parameters map; we will need to set the nodeID parameter to correspond to the node in the temporary scene
+    qSlicerIO::IOProperties temporarySceneParameters = parameters;
     temporarySceneParameters["nodeID"] = temporaryStorableNode->GetID();
-    temporarySceneParameters["fileName"] = filePaths[i];
 
     // Deduce "fileType" from "fileFormat" parameter; saveNodes will want both
-    qSlicerIO::IOFileType fileType = this->fileWriterFileType(storableNode, parameters.value("fileFormat").toString());
+    qSlicerIO::IOFileType fileType = this->fileWriterFileType(storableNode, parameters["fileFormat"].toString());
+
+    // Add default storage node into the temporary scene. This is sometimes needed.
```

Comment:
This is the part that fixed the following bug I mentioned in my summary:
> A mild export-related bug has been fixed where plane markups would get written by `vtkMRMLMarkupsJsonStorageNode` rather than `vtkMRMLMarkupsPlaneJsonStorageNode`, and similarly for ROIs. This caused them to be written incorrectly.